src/main/java/com/google/sps/data/User.java

```diff
@@ -1,57 +1,198 @@
+        testUser.entity = new Entity(userKey);
+        String emptyJsonArray = "[ ]";
+        testUser.entity.setProperty("displayName", "Test Person");
+        testUser.entity.setProperty("cookbook", emptyJsonArray);
+        testUser.entity.setProperty("userRecipes", emptyJsonArray);
+        testUser.entity.setProperty("userRecipes", emptyJsonArray);
+        testUser.entity.setProperty("planner", emptyJsonArray);
+    }
+    return testUser;
+  }
+
+  public long getDisplayName() {
+    return (long) entity.getProperty("displayName");
+  }
+
+  public void setDisplayName(String name) {
+    entity.setProperty("displayName", name);
+    putEntity();
+  }
+
+// Cookbook and Planner currently support only one type of Recipe entity (namely public).
+// Likewise, userRecipes should only include private recipes.
+
+  public ArrayList<Long> getCookbook() {
+    return getPropertyArrayList("cookbook");
   }
 
-  public void setId(String id) {
-    this.id = id;
+  public ArrayList<Long> getUserRecipes() {
+    return getPropertyArrayList("userRecipes");
   }
 
-  public boolean equalIds(String id) {
-    return this.id.equals(id);
+  public ArrayList<Long> getPlanner() {
+    return getPropertyArrayList("planner");
   }
 
   public void addRecipeToPlanner(long id) {
-    planner.add(id);
+    addIDToRecipeList(id, "planner");
   }
 
   public void addRecipeToCookbook(long id) {
-    cookbook.add(id);
+    addIDToRecipeList(id, "cookbook");
   }
 
   public void addRecipeToUserRecipes(long id) {
-    userRecipes.add(id);
+    addIDToRecipeList(id, "userRecipes");
   }
 
   public void removeRecipeFromPlanner(long id) {
-    planner.remove(new Long(id));
+    removeIDFromRecipeList(id, "planner");
   }
 
   public void removeRecipeFromCookbook(long id) {
-    cookbook.remove(new Long(id));
+    removeIDFromRecipeList(id, "cookbook");
   }
 
   public void removeRecipeFromUserRecipes(long id) {
-    userRecipes.remove(new Long(id));
+    removeIDFromRecipeList(id, "userRecipes");
+  }
+
+  /**
+   * Helper method for removing recipe IDs form specific user recipe list.
+   * @Param id recipe ID to be removed from current User entity.
+   * @Param recipeListType Property name of the list to removed from.
+   */
+  private void removeIDFromRecipeList(long id, String recipeListType) {
+    Gson gson = new Gson();
+    ArrayList<Long> list = getPropertyArrayList(recipeListType);
+    list.remove(new Long(id));
+    entity.setProperty(recipeListType, gson.toJson(list));
+    putEntity();
+  }
+
+  /**
+   * Helper method for adding recipe IDs to specific user recipe list.
+   * @Param id recipe ID to be added to current User entity.
+   * @Param recipeListType Property name of the list to add to.
+   */
+  private void addIDToRecipeList(long id, String recipeListType) {
+    Gson gson = new Gson();
+    ArrayList<Long> list = getPropertyArrayList(recipeListType);
+    if (!list.contains(id)) {
+      list.add(id);
+    }
```

Comment:
Have you considered using a hash set? You could avoid doing this check, and it has the added benefit of a constant time remove in the above method. Using a hash set doesn't guarantee an iteration order though, if that is something you're dependent on.

Reply:
I will definitely consider this for a future pr, I'll probably revise for efficiency once I at least have the user verification working properly.

Reply:
Sure that sounds good